shell.nix

```diff
@@ -20,72 +20,23 @@ with builtins;
 let
   inherit (pkgs) stdenv lib;
 
-  # Use builtins.fromTOML if available, otherwise use remarshal to
-  # generate JSON which can be read. Code taken from
-  # nixpkgs/pkgs/development/tools/poetry2nix/poetry2nix/lib.nix.
-  fromTOML = pkgs: builtins.fromTOML or (
-    toml: builtins.fromJSON (
-      builtins.readFile (
-        pkgs.runCommand "from-toml"
-          {
-            inherit toml;
-            allowSubstitutes = false;
-            preferLocalBuild = true;
-          }
-          ''
-            ${pkgs.remarshal}/bin/remarshal \
-              -if toml \
-              -i <(echo "$toml") \
-              -of json \
-              -o $out
-          ''
-      )
-    )
-  );
-
-  # Tockloader v1.10.0
+  # Tockloader v1.11.0
   tockloader = import (pkgs.fetchFromGitHub {
     owner = "tock";
     repo = "tockloader";
-    # TODO: change to tag once there is a Tockloader release with
-    # `default.nix` included.
-    rev = "6f37412d5608d9bb48510c98a929cc3f96f8cc8f";
-    sha256 = "sha256-0WobupjSqJ36+nME9YO9wcEx4X6jE+edSn4PNM+aDUo=";
+    rev = "v1.11.0";
+    sha256 = "sha256-bPEfpfOZOjOiazqRgn1cnqe4ohLPvocuENKoZx/Qw80=";
   }) { inherit pkgs withUnfreePkgs; };
 
-  moz_overlay = import (builtins.fetchTarball https://github.com/mozilla/nixpkgs-mozilla/archive/master.tar.gz);
-  nixpkgs = import <nixpkgs> { overlays = [ moz_overlay ]; };
+  rust_overlay = import ("${fetchTarball "https://github.com/nix-community/fenix/archive/main.tar.gz"}/overlay.nix");
```

Comment:
While we're thinking about dependency security here, would it be possible for you to change this to a `builtins.fetchGit` invocation with a fixed commit hash? This way we're isolated from breaking / potentially malicious changes upstream.

This has the drawback that, with `fenix`, our periodic updates to `rust-toolchain.toml` are guaranteed to break the Nix expression, but on the other hand this will allow us to evaluate this expression offline, from cached versions in the Nix store.